### DISCUSSION TITLE
How to stop the execution of the server event handler if the request is aborted or destroyed or canceled ?

### DISCUSSION THREAD

**Discussion by XStarlink** [nuxt/nuxt#22535] [Questions]
Hello, I'm using Nuxt 3.6.5, I've searched everywhere but I can't find a way to detect server side if the request is aborted or destroyed or canceled by the user.

Currently, when I launch a request (which performs long processing) and click on a button that triggers an AbortControler, the front end stops receiving the stream and terminates the request, but my server doesn't detect the request abort and continues processing until the end. 

I'd like to be able to detect the cancellation of the request on the server side and cleanly kill the processing that's being done.

I've tried many things in server like:
```js
event.node.req.socket.on('close', () => {
      console.log('CLOSED')
})

event.node.req.socket.on('end', () => {
      console.log('END')
})

 event.node.req.connection.on('close', () => {
      console.log('CLOSED')
})

event.node.req.addListener('abort', () => {
      console.log('ABORTED')
})

event.node.req.addListener('close', () => {
      console.log('CLOSED')
})

 event.node.req.on('error', () => {
      console.log('ERROR')
})

console.log('Destroyed', event.node.req.destroyed)
```
I did a lot of tests but nothing worked, If anyone can give me some advice, that would be really great.

Thanks in advance

**Comment by gorango** [comment]
listening to **`res.on('close')`** works:

```js
  const controller = new AbortController();
  event.node.res.on("close", () => {
    controller.abort();
  });
  ```
  
https://github.com/nitrojs/nitro/issues/2118
